### DISCUSSION TITLE
before_send not scrubbing all headers/query strings

### DISCUSSION THREAD

**Discussion by robbieallen-fareharbor** [getsentry/sentry-python#1753] [Q&A] [answered]
Hello, I recently implemented a `before_send` to scrub certain keys before sending into Sentry. When I went back into Sentry to verify the keys were scrubbed they still show up for certain `INFO` issues (N+1 query). On all other errors, they are scrubbed out as expected.

Is there a reason why the N+1 issues would still show the keys in the query strings/headers? When I tested by manually adding an `INFO` issue with the keys in the headers and query strings then they are scrubbed out, so I wasn't thinking it's just an `INFO` status issue.

I am filtering with python sentry_sdk module and any feedback is greatly appreciated!

PS: There is a similar topic about python sentry_sdk not filtering [here](https://github.com/getsentry/sentry-python/discussions/1682)

**Comment by antonpirker** [comment, marked as answer]
Hey @robbieallen-fareharbor,  thanks for reporting this!

This is a special edge case. 

The N+1 issues are not sent by the SDK but are created in Sentry.io by the performance data the SDK sends. The performance data (called transactions) is not filtered through `beforeSend`. We are currently work on a `beforeSendTransaction` where you can filter fields from transactions. But it is not there yet. The progress can be watched here: https://github.com/getsentry/sentry-python/issues/1738
